### PR TITLE
Upgrade Rust to rustc 1.5.0-nightly (b2f379cdc 2015-09-23)

### PR DESCRIPTION
**Pull request by pietro**
Ident was removed in many HIR structures in favor of Name.

### CODE REVIEW

File: src/len_zero.rs [hunk 1]
```diff
@@ -101,17 +102,17 @@ fn check_cmp(cx: &LateContext, span: Span, left: &Expr, right: &Expr, op: &str)
...
         _ => ()
     }
 }
 
-fn check_len_zero(cx: &LateContext, span: Span, method: &SpannedIdent,
+fn check_len_zero(cx: &LateContext, span: Span, name: &Name,
                   args: &[P<Expr>], lit: &Lit, op: &str) {
     if let Spanned{node: LitInt(0, _), ..} = *lit {
-        if method.node.name == "len" && args.len() == 1 &&
+        if name.as_str() == "len" && args.len() == 1 &&
```
Comment (llogiq) [inline src/len_zero.rs]: Is the .as_str() necessary here?

File: src/methods.rs [hunk 2]
```diff
@@ -40,9 +40,9 @@ impl LintPass for MethodsPass {
 
 impl LateLintPass for MethodsPass {
     fn check_expr(&mut self, cx: &LateContext, expr: &Expr) {
-        if let ExprMethodCall(ref ident, _, ref args) = expr.node {
+        if let ExprMethodCall(ref name, _, ref args) = expr.node {
             let (obj_ty, ptr_depth) = walk_ptrs_ty_depth(cx.tcx.expr_ty(&args[0]));
-            if ident.node.name == "unwrap" {
+            if name.node.as_str() == "unwrap" {
```
Comment (llogiq) [inline src/methods.rs]: Same here

File: src/methods.rs [hunk 3]
```diff
@@ -54,7 +54,7 @@ impl LateLintPass for MethodsPass {
                                of Err values is preferred");
                 }
             }
-            else if ident.node.name == "to_string" {
+            else if name.node.as_str() == "to_string" {
```
Comment (llogiq) [inline src/methods.rs]: And here.

File: src/misc.rs [hunk 4]
```diff
@@ -173,10 +173,9 @@ impl LateLintPass for CmpOwned {
 
 fn check_to_owned(cx: &LateContext, expr: &Expr, other_span: Span) {
     match expr.node {
-        ExprMethodCall(Spanned{node: ref ident, ..}, _, ref args) => {
-            let name = ident.name;
-            if name == "to_string" ||
-                name == "to_owned" && is_str_arg(cx, args) {
+        ExprMethodCall(Spanned{node: ref name, ..}, _, ref args) => {
+            if name.as_str() == "to_string" ||
+                name.as_str() == "to_owned" && is_str_arg(cx, args) {
```
Comment (llogiq) [inline src/misc.rs]: And doubly here.

File: src/ranges.rs [hunk 5]
```diff
@@ -19,10 +19,10 @@ impl LintPass for StepByZero {
 
 impl LateLintPass for StepByZero {
     fn check_expr(&mut self, cx: &LateContext, expr: &Expr) {
-        if let ExprMethodCall(Spanned { node: ref ident, .. }, _,
+        if let ExprMethodCall(Spanned { node: ref name, .. }, _,
                               ref args) = expr.node {
             // Only warn on literal ranges.
-            if ident.name == "step_by" && args.len() == 2 &&
+            if name.as_str() == "step_by" && args.len() == 2 &&
```
Comment (llogiq) [inline src/ranges.rs]: And another one.

**Comment by Manishearth** [comment]
(fixed the nits before merging)

**Pull request merged**